<?xml version="1.0" ?>
<robot name="r32" xmlns:xacro="http://ros.org/wiki/xacro">
    <xacro:include filename="$(find uwb_v2v_ros_sim)/rail_vehicle_description/model.gazebo.xacro"/>
    <xacro:include filename="$(find uwb_v2v_ros_sim)/rail_vehicle_description/model.uwb.xacro"/>

    <link name="base_footprint"/>

    <joint name="base_joint" type="fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
        <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
    </joint>

    <link name="base_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://uwb_v2v_ros_sim/rail_vehicle_description/meshes/r32.dae" scale="1 1 1"/>
            </geometry>
        </visual>

        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://uwb_v2v_ros_sim/rail_vehicle_description/meshes/r32.dae" scale="1 1 1"/>
            </geometry>
        </collision>

        <inertial>
            <origin xyz="0 0 1" rpy="0 0 0"/>
            <mass value="20000"/>
            <inertia ixx="4.4166e05" ixy="0" ixz="0" iyy="3e04" iyz="0" izz="4.4166e05"/>
        </inertial>
        <dynamics damping="0.1" friction="0.1"/>
    </link>

    <xacro:A_end_virtual_frame half_vehicle_length="7.1"/>
    <xacro:uwb_link link_name="uwb_slave_A"
                    visual_origin="0 0 0" collision_origin="0 0 0" geometry_size="0.1 0.1 0.2"
                    joint_origin="1.2 0.0 4.0" parent_name="A_end_virtual_frame"/>
    <xacro:uwb_link link_name="uwb_master_A"
                    visual_origin="0 0 0" collision_origin="0 0 0" geometry_size="0.1 0.1 0.2"
                    joint_origin="-1.2 0.0 4.0" parent_name="A_end_virtual_frame"/>
    <xacro:B_end_virtual_frame half_vehicle_length="7.1"/>
    <xacro:uwb_link link_name="uwb_slave_B"
                    visual_origin="0 0 0" collision_origin="0 0 0" geometry_size="0.1 0.1 0.2"
                    joint_origin="1.2 0.0 4.0" parent_name="B_end_virtual_frame"/>
    <xacro:uwb_link link_name="uwb_master_B"
                    visual_origin="0 0 0" collision_origin="0 0 0" geometry_size="0.1 0.1 0.2"
                    joint_origin="-1.2 0.0 4.0" parent_name="B_end_virtual_frame"/>

</robot>
